examples/vision/fixres.py

```diff
@@ -0,0 +1,319 @@
+
+
+"""
+Notice how the augmentation transforms vary for the kind of dataset we are preparing.
+"""
+
+"""
+## Prepare datasets
+"""
+
+initial_train_dataset = make_dataset(train_dataset, train=True, image_size=smaller_size)
+initial_val_dataset = make_dataset(val_dataset, train=False, image_size=smaller_size)
+
+finetune_train_dataset = make_dataset(train_dataset, train=True, image_size=bigger_size)
+finetune_val_dataset = make_dataset(val_dataset, train=False, image_size=bigger_size)
+
+vanilla_train_dataset = make_dataset(
+    train_dataset, train=True, image_size=bigger_size, fixres=False
+)
+vanilla_val_dataset = make_dataset(
+    val_dataset, train=False, image_size=bigger_size, fixres=False
+)
+
+"""
+## Visualize the datasets
+"""
+
+
+def visualize_dataset(batch_images):
+    plt.figure(figsize=(10, 10))
+    for n in range(25):
+        ax = plt.subplot(5, 5, n + 1)
+        plt.imshow((batch_images[n].numpy() + 1) / 2)
+        plt.axis("off")
+    plt.show()
+
+    print(f"Batch shape: {batch_images.shape}.")
+
+
+# Smaller resolution.
+initial_sample_images, _ = next(iter(initial_train_dataset))
+visualize_dataset(initial_sample_images)
+
+# Bigger resolution but only for fine-tuning purposes.
+finetune_sample_images, _ = next(iter(finetune_train_dataset))
+visualize_dataset(finetune_sample_images)
+
+# Bigger resolution with the same augmentation applied as
+# the smaller resolution dataset.
+vanilla_sample_images, _ = next(iter(vanilla_train_dataset))
+visualize_dataset(vanilla_sample_images)
+
+"""
+## Model training utilities
+
+We will train multiple variants of ResNet50V2
+([He et al.](https://arxiv.org/abs/1603.05027)):
+
+1. On the smaller resolution dataset (128x128). It will be trained from scratch.
+2. Then fine-tune the model from 1 on the larger resolution (224x224) dataset. 
+3. Train another ResNet50V2 from scratch on the larger resolution dataset. 
+
+As a reminder, the larger resolution datasets will differ in terms of the augmentation
+transforms.
+"""
+
+
+def get_training_model(num_classes=5):
+    inputs = layers.Input((None, None, 3))
+    resnet_base = keras.applications.ResNet50V2(
+        include_top=False, weights=None, pooling="avg"
+    )
+    resnet_base.trainable = True
+
+    x = resnet_base(inputs)
+    outputs = layers.Dense(num_classes, activation="softmax")(x)
+    return keras.Model(inputs, outputs)
+
+
+def train_and_evaluate(model, datasets, epochs, lr=None, use_es=False):
+    if lr:
+        optimizer = keras.optimizers.Adam(learning_rate=lr)
+    else:
+        optimizer = "adam"
+
+    model.compile(
+        optimizer=optimizer,
+        loss="sparse_categorical_crossentropy",
+        metrics=["accuracy"],
+    )
+
+    if use_es:
+        es_callback = keras.callbacks.EarlyStopping(patience=5)
+        callbacks = [es_callback]
+    else:
+        callbacks = None
+
+    training_dataset = datasets[0]
+    validation_dataset = datasets[1]
```

Comment:
Making these 2 separate arguments in the function signature would be more readable